pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -146,14 +146,14 @@ var labelReconcileInfo = []struct {
 		// TODO(#89477): no earlier than 1.22: drop the beta labels if they differ from the GA labels
 		primaryKey:            v1.LabelOSStable,
 		secondaryKey:          kubeletapis.LabelOS,
-		ensureSecondaryExists: true,
+		ensureSecondaryExists: false,
 	},
 	{
 		// Reconcile the beta and the stable arch label using the stable label as the source of truth.
 		// TODO(#89477): no earlier than 1.22: drop the beta labels if they differ from the GA labels
 		primaryKey:            v1.LabelArchStable,
 		secondaryKey:          kubeletapis.LabelArch,
-		ensureSecondaryExists: true,
+		ensureSecondaryExists: false,
```

Comment:
I'm not convinced we've made use of the beta labels sufficiently visible to users and given them the information they need to transition to the GA labels successfully. See my comment in https://github.com/kubernetes/kubernetes/issues/89477#issuecomment-834592711

Reply:
Agree.
Since the cost of maintaining them is relatively low, keeping it for several more releases would be more friendly to end users.